Golem messages

kbeker opened the pull request:
Golem-messages decoder pull request

cameel commented on concent_api/utils/api_view.py:
```diff
@@ -0,0 +1,59 @@
+                    request.body,
+                    settings.CONCENT_PRIVATE_KEY,
+                    client_public_key
+            )
```
Please always run `./full-check.py` before submitting a pull request. This like generates a warning:

```
./concent_api/utils/api_view.py:36:13: E122 continuation line missing indentation or outdented
```

cameel commented on concent_api/requirements.txt:
```diff
@@ -1,2 +1,3 @@
 Django >= 1.11, < 1.12
 psycopg2
+-e git://github.com/golemfactory/golem-messages.git@master#egg=golem_messages
```
I see there are tagged releases now so let's switch from `master` to `v1.1.0`.

Don't forget to regenerate `concent_api/requirements.lock` after you change that.

cameel commented on concent_api/requirements.lock:
```diff
@@ -1,3 +1,24 @@
+cffi==1.11.2
+coincurve==6.0.0
 Django==1.11.7
+freezegun==0.3.9
```
`golem-messages` does not depend on `freezegun`,  `python-dateutil` and `six`. These packages should not be listed here. Please regenerate `requirements.lock` on a clean virtualenv to make sure you have no stray packages here.

cameel commented on concent_api/utils/api_view.py:
```diff
@@ -0,0 +1,59 @@
+            if request.content_type == '':
+                raise Http400('Content-Type is missing.')
+            if request.content_type == 'application/json':
+                message = request.body
```
You should deserialize the JSON message before passing it to the view.

cameel commented on concent_api/utils/tests/test_api_view.py:
```diff
@@ -0,0 +1,126 @@
+
+        dummy_view(request)  # pylint: disable=no-value-for-parameter
+        self.assertIsInstance(decoded_message, MessageWantToComputeTask)
+        self.assertEqual(decoded_message.node_name,         self.message_want_to_compute.node_name)
```
You should create a function that converts the message to a dict and compares to those dicts instead of listing every field individually as we agreed yesterday. This would reduce repetition here.

kbeker commented:
I've added deserialization in api_view, now view gets dict when content-type is application/json. I've also added function to convert message into dict in test_api_view. Regenerated requirements.txt and requirements.lock. Checked on ./full-check.py

cameel commented:
> 2) There's no unit test for the case where the view gets a JSON message (and you have a bug there that went undetected; see my other comments).

You missed this one. You still don't have a test for that case.

cameel commented on concent_api/utils/api_view.py:
```diff
@@ -20,20 +20,19 @@ def wrapper(request, *args, **kwargs):
                 raise Http400('Content-Type is missing.')
             if request.content_type == 'application/json':
-                message = request.body
+                message = json.loads(request.body.decode('ascii').replace("'", "\""))
```
Why are you replacing single quotes with double quotes in the message? This does not look like something that belongs here.

kbeker commented:
Add missed test - test_api_view_view_should_get_json

kbeker commented:
Fixed sending in tests and getting in api_view json

kbeker commented:
Changed header CONCENT_CLIENT_PUBLIC_KEY to HTTP_CONCENT_CLIENT_PUBLIC_KEY

cameel commented:
There were still some small issues, mostly with the code in README but I fixed them myself. The code can be merged now.
